### DISCUSSION TITLE
Shape sub type

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#71] [Q&A] [answered]
I have a primitive convex shape, e.g. `EShapeSubType_Box`. If I have a body, I can get it by `body.GetShape().GetSubType()`. However, if I wrap it in a scaled shape, e.g.  `new Jolt.ScaledShapeSettings(settings, vec);`, then I will no longer be able to know that its a box. The sub type would become `EShapeSubType_Scaled`. Is there another way to learn the underlying shape of a scaled shape?

**Comment by jrouwe** [comment]
The ```GetInnerShape()``` function was missing. I've added it now.

**Reply by LeXXik** [reply]
Thank you! I can see it is now available on the shape settings. I am not sure how to get it from the body, though. Is there a method that exposes shape settings from body?

Edit:
This is not really important. I was just doing a test case, where I needed to know the shape type of a colliding body. I no longer have this test, though.

**Reply by jrouwe** [reply, marked as answer]
It's a member of ScaledShape, you need to cast your body's shape to ScaledShape if you detect that the GetSubType() is a scaled shape then you can test GetInnerShape().GetSubType() (which may also not be a leaf shape, so you need to write a recursive function).

**Reply by LeXXik** [reply]
Thank you!
